R/scenario.R

```diff
@@ -47,151 +53,137 @@ new_scenario <- function(model_function,
+  extra_info <- glue::glue_collapse(
+    glue::glue("'{names(x$extra_info)}'"),
+    sep = ", "
+  )
+  extra_info <- cli::col_cyan(extra_info)
+
+  # print to screen
   writeLines(
     c(
-      sprintf("Epidemic scenario object"),
-      sprintf(" Model function: %s", x$model_function),
-      sprintf(" Scenario replicates: %s", x$replicates),
-      sprintf(" Scenario outcomes are %s", ifelse(
-        sce_has_data(x), "prepared", "not prepared"
-      ))
+      cli::style_bold("Epidemic scenario object"),
+      glue::glue(" Model function: {cli::col_cyan(x$model_function)}"),
+      glue::glue(" Extra information on: {extra_info}"),
+      glue::glue(" Scenario replicates: {x$replicates}"),
+      glue::glue(
+        "
+         Scenario outcomes are \\
+        {ifelse(sce_has_data(x), 'prepared', 'not prepared')}
+        "
+      )
     )
   )
 
   invisible(x)
 }
 
-#' Run a scenario
+#' Check whether an object is a 'scenario'
 #'
-#' @description Run an epidemic model scenario using the function stored in
-#' `model_function` with the arguments in `parameters`. Runs as many replicates
-#' of the function as specified in `replicates`. The simulation output data are
-#' stored under `data`, and the `data_available` tag is updated to `TRUE`.
+#' @param x An R object.
 #'
-#' @param x A `scenario` object.
+#' @return A logical indicating whether the object inherits from the class
+#' 'scenario'.
 #'
-#' @return The original `scenario` object with the `data` field populated with
-#' simulation output. This object must be assigned.
 #' @export
 #'
 #' @examples
-#' # prepare a scenario
-#' scenario_pandemic_flu <- scenario(
+#' # prepare two scenarios of the final size of an epidemic
+#' pandemic_flu <- scenario(
 #'   model_function = "finalsize::final_size",
-#'   parameters = make_parameters_finalsize_UK(), # using helper function
+#'   parameters = make_parameters_finalsize_UK(r0 = 1.5),
 #'   replicates = 1L
 #' )
 #'
-#' # print to check that data are not prepared
-#' scenario_pandemic_flu
-#'
-#' # generate scenario data
-#' scenario_pandemic_flu <- run_scenario(scenario_pandemic_flu)
-#'
-#' # print to check that data are prepared
-#' scenario_pandemic_flu
-run_scenario <- function(x) {
-
-  # input checking
-  checkmate::assert_class(x, "scenario")
-
-  # get model function whether explicitly namespaced or not
-  if (grepl(pattern = "::", x = x$model_function, fixed = TRUE)) {
-    pkg_name <- sub("::.*", "", x$model_function)
-    fn_name <- sub(".*::", "", x$model_function)
-    fn <- get(fn_name, asNamespace(pkg_name))
-  } else {
-    fn <- x$model_function
-  }
-
-  # populate the data list
-  output <- lapply(
-    seq_len(x$replicates),
-    function(i) {
-      data <- do.call(
-        what = fn,
-        args = x$parameters
-      )
-      data$replicate <- i
-      data
-    }
-  )
-  # assign output to data
-  x$data <- output
-
-  x
-}
+#' is.scenario(pandemic_flu)
+is.scenario <- function(x) inherits(x, "scenario")
```

Comment:
`is_scenario()` for consistent naming

Reply:
Fixed in https://github.com/epiverse-trace/scenarios/pull/12/commits/4401b480777a73ad1c65b137dadcf42ce00a3650